client/extensions/woocommerce/app/products/product-variation-types-form.js

```diff
@@ -31,13 +31,19 @@ export default class ProductVariationTypesForm extends Component {
 		if ( ! product.attributes ) {
 			this.addType();
 		}
+
+		this.debouncedUpdateName = debounce( this.updateName, 300 );
 	}
 
 	constructor( props ) {
 		super( props );
 
+		this.state = {
+			attributeNames: [],
```

Comment:
Just curious here. You seems to be using `attributeNames` as an object. I know its valid, but why not just use an Object? Is it because you can avoid mutating state by easily cloning it via `[ ... attributeNames ]`?